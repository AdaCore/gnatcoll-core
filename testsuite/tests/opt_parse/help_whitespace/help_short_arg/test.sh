STDOUT=stdout.log

./test --help > $STDOUT

# Check that the help message is structured as expected
if ! grep -q -P "( ){4,}" $STDOUT; then
    # The biggest whitespace should be the three characters before optional
    # arguments as "--help, -h" and "--char, -C" are the same lenght, and
    # while they are less than 25 characters, the column number the help
    # text starts at should be limited to 2 after the longest arg text.
    echo "Help message column reduced as expected"
fi
