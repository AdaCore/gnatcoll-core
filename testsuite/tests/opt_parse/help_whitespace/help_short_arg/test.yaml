driver: build_run_diff
description: Check that long argument name causes the previously default 25 column limit to shrink