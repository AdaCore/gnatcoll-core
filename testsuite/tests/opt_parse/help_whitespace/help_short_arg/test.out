Help message column reduced as expected